FeedStoreChallenge/CacheEntity+CoreDataProperties.swift

```diff
@@ -0,0 +1,39 @@
+//
+//  CacheEntity+CoreDataProperties.swift
+//  FeedStoreChallenge
+//
+//  Created by Mohammed Alwaili on 02/03/2021.
+//  Copyright © 2021 Essential Developer. All rights reserved.
+//
+//
+
+import Foundation
+import CoreData
+
+extension CacheEntity {
+
+    @nonobjc class func fetchRequest() -> NSFetchRequest<CacheEntity> {
+        return NSFetchRequest<CacheEntity>(entityName: "CacheEntity")
+    }
+
+    @NSManaged var timestamp: Date
+    @NSManaged var feed: NSOrderedSet
+
+}
+
+extension CacheEntity {
+
+	static func newInstance(in context: NSManagedObjectContext) throws -> CacheEntity {
+		let request: NSFetchRequest<CacheEntity> = CacheEntity.fetchRequest()
+		if let cache = try context.fetch(request).first {
```

Comment:
There's duplication here. The logic is exactly the same as in the `fetch(in context: NSManagedObjectContext)` method. So you can use it instead of duplicating the code.

```suggestion
	static func newInstance(in context: NSManagedObjectContext) throws -> CacheEntity {
		if let cache = try fetch(in: context) {
```